packages/nimble-components/src/table/specs/table-column-sort-hld.md

```diff
@@ -0,0 +1,77 @@
+
+In some cases, a client may want to configure their table to be sorted by a column that is not visible within the table. To enable this, a column within the table can be marked as hidden. A hidden column exists in the table to provide metadata that can be used for operations, such as sorting, but it is not rendered within the table.
+
+A column within the table may not have a one-to-one mapping to a field within the table's data. For example, a hyperlink column might use two fields from the data: a string to use as the hyperlink's display and a string to use as the hyperlink's href. Because of this, a column must define which data field is used when sorting that column. That field is specified by a column definition by implementing an `operationalDataFieldName` getter that returns the name of the field to use for sorting. For example, the text column would return the value of `fieldName` and the hyperlink column would return the field name associated with the display string.
+
+The column definition will also be responsible for specifying the appropriate sort function to use based on the type of data it knows is associated with `operationalDataFieldName`. The sort function will be specified by a `sortFunction` getter on the column, that returns a `TableColumnSortFunction` value. `TableColumnSortFunction` is a new enum that will be introduced that defines the set of sort functions that can be applied to a column. Initially, the enum will consist of the value `basic`, which will correspond to [TanStack's `basic` sort function](https://tanstack.com/table/v8/docs/api/features/sorting#sorting-functions), and the value `localeAwareCaseSensitive`, which will be backed by a function written within nimble. That function will leverage [localeCompare()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare) to perform the sort.
+
+For performance reasons, the table will not support custom sort functions on a column. Additionally, a column can only be sorted based on a single field within the table's data. These requirements guard against performance degradations from inefficient custom sort functions.
+
+#### Summary of new attributes on the base table column
```

Comment:
Can you describe more precisely the plugin author's api they need to implement?

Right now it sounds like they would implement a `get operationalDataFieldName()` and a `get sortFunction`.
I think to align with our `getColumnConfig` and `getDataRecordFieldNames` methods maybe this should instead be `getOperationalDataFieldName` and `getSortFunction`.

Reply:
Maybe I'm mistaken because I haven't actually tried it out, but I'm pretty sure that we can't rely on the FAST's observables and the existing notification pattern if we make `getOperationalDataFieldName` and `getSortFunction` methods rather than properties. I envision us wanting to listen to changes to those values so we can re-run the sort.

FYI @atmgrifter00, I think you were planning on trying to remember/find the rationale behind making those other members functions rather than properties. Did you get a chance to do that?

Reply:
@mollykreis I haven't but I will now go and try and do that.

Reply:
The plan is to use properties for everything in the base class so that we ensure everything is observable. #1060